client/src/test/suite/extension.test.ts

```diff
@@ -306,4 +323,42 @@ suite('Extension Test Suite', function (this: Suite) {
     populateXsdSchemaFiles(TEST_OUT_DIR)
     assert(!fs.existsSync(testXsdPath))
   })
+  test('getRepo returns repository', async () => {
+    const repo = pushContent.getRepo()
+    assert.notStrictEqual(repo.rootUri, undefined)
+  })
+  test('push with no conflict', async () => {
+    stubRepo.commit = sinon.stub().throws()
+    stubRepo.pull = sinon.stub().returns(Promise.resolve())
+    stubRepo.push = sinon.stub().returns(Promise.resolve())
+    await pushContent._pushContent(getRepo, reporter('Successful content push.'), ignore)
+  })
+  test('push with merge conflict', async () => {
+    stubRepo.commit = sinon.stub().returns(Promise.resolve())
+    stubRepo.pull = sinon.stub().throws(() => {
+      const error: any = new Error()
+      error.gitErrorCode = GitErrorCodes.Conflict
+      return error
+    })
+    stubRepo.push = sinon.stub().returns(Promise.resolve())
+    try {
+      await pushContent._pushContent(getRepo, ignore, reporter('Content conflict, please resolve.'))
+    } catch (e) {
+      assert.notStrictEqual(e.gitErrorCode, GitErrorCodes.Conflict)
+    }
+  })
+  test('push with no changes', async () => {
+    stubRepo.commit = sinon.stub().throws(() => {
+      const error: any = new Error()
+      error.stdout = 'nothing to commit '
+      return error
+    })
+    stubRepo.pull = sinon.stub().returns(Promise.resolve())
+    stubRepo.push = sinon.stub().returns(Promise.resolve())
+    try {
+      await pushContent._pushContent(getRepo, ignore, reporter('No changes to push.'))
```

Comment:
you might want to use `assert.doesNotReject` here

Reply:
And careful, `pushContent.__pushContent(...)` returns an async function, not a promise, awaiting it doesn't do anything. Luckily,  `assert.doesNotReject` does allow you to pass in an async function taking no arguments.